evm/src/generation/partial_trie.rs

```diff
@@ -0,0 +1,24 @@
+use ethereum_types::U256;
+
+/// A partial trie, or a sub-trie thereof.
+pub enum PartialTrie {
+    /// An empty trie.
+    Empty,
+    /// The digest of trie whose data does not need to be stored.
+    Hash(U256),
+    /// A branch node, which consists of 16 children and an optional value.
+    Branch([Box<PartialTrie>; 16], Option<U256>),
+    /// An extension node, which consists of a list of nibbles and a single child.
+    Extension(Nibbles, Box<PartialTrie>),
+    /// A leaf node, which consists of a list of nibbles and a value.
+    Leaf(Nibbles, Vec<u8>),
```

Comment:
nit: Maybe it would be clearer to use named fields here instead of tuples. Like
```rust
Branch { children: [Box<PartialTrie>; 16], value: Option<U256>},
Extension { nibbles: Nibbles, child: Box<PartialTrie>},
Leaf { nibbles: Nibbles, value: Vec<u8>}
```

Reply:
Good idea 👍